contracts/ServiceManagerToken.sol

```diff
@@ -23,29 +31,47 @@ interface IToken {
     function ownerOf(uint256 tokenId) external view returns (address);
 }
 
+/// @dev The contract is already initialized.
+error AlreadyInitialized();
+
 /// @title Service Manager - Periphery smart contract for managing services with custom ERC20 tokens or ETH
 /// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
-/// @author AL
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
 contract ServiceManagerToken is GenericManager, OperatorSignedHashes {
     event OperatorWhitelistUpdated(address indexed operatorWhitelist);
+    event IdentityRegistryBridgerUpdated(address indexed identityRegistryBridger);
+    event ImplementationUpdated(address indexed implementation);
     event CreateMultisig(address indexed multisig);
+    event ServiceAgentRegisteredOrUpdated(uint256 indexed serviceId, uint256 indexed agentId, bool registered);
+
+    // Name of a signing domain
+    string public constant NAME = "OLAS Service Manager";
+    // Version number
+    string public constant VERSION = "1.2.0";
+    // Service Manager proxy address slot
+    // keccak256("PROXY_SERVICE_MANAGER") = "0xe39e69948a448ce9239ad71b908b6c5b46225f86ffa735b25a8cd64080315855"
+    bytes32 public constant PROXY_SERVICE_MANAGER = 0xe39e69948a448ce9239ad71b908b6c5b46225f86ffa735b25a8cd64080315855;
+    // A well-known representation of ETH as an address
+    address public constant ETH_TOKEN_ADDRESS = address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+    // Bond wrapping constant
+    uint96 public constant BOND_WRAPPER = 1;
 
     // Service Registry address
     address public immutable serviceRegistry;
     // Service Registry Token Utility address
     address public immutable serviceRegistryTokenUtility;
-    // A well-known representation of ETH as an address
-    address public constant ETH_TOKEN_ADDRESS = address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
-    // Bond wrapping constant
-    uint96 public constant BOND_WRAPPER = 1;
+
+    // 8004 Identity Registry Bridger address
+    address public identityRegistryBridger;
     // Operator whitelist address
     address public operatorWhitelist;
 
-    /// @dev ServiceRegistryTokenUtility constructor.
-    /// @param _serviceRegistry Service Registry contract address.
-    /// @param _serviceRegistryTokenUtility Service Registry Token Utility contract address.
-    constructor(address _serviceRegistry, address _serviceRegistryTokenUtility, address _operatorWhitelist)
-        OperatorSignedHashes("Service Manager Token", "1.1.1")
+    /// @dev ServiceManager constructor.
+    /// @param _serviceRegistry Service Registry address.
+    /// @param _serviceRegistryTokenUtility Service Registry Token Utility address.
+    constructor(address _serviceRegistry, address _serviceRegistryTokenUtility)
+        OperatorSignedHashes(NAME, VERSION)
```

Comment:
Just a small optimization - don't hardcode domain name and version in the constructor, but rather have it as constants.